core/mavlink_system.cpp

```diff
@@ -334,20 +334,42 @@ size_t MAVLinkSystem::total_components() const
 
 bool MAVLinkSystem::is_standalone() const
 {
-    return !is_autopilot();
+    return !has_autopilot();
 }
 
-bool MAVLinkSystem::is_autopilot() const
+bool MAVLinkSystem::has_autopilot() const
 {
     return get_autopilot_id() != uint8_t(0);
 }
 
+bool MAVLinkSystem::is_autopilot(uint8_t comp_id)
+{
+    return comp_id == MAV_COMP_ID_AUTOPILOT1;
+}
+
+bool MAVLinkSystem::is_camera(uint8_t comp_id)
+{
+    return (comp_id >= MAV_COMP_ID_CAMERA)
+           && (comp_id <= MAV_COMP_ID_CAMERA6);
+}
+
 bool MAVLinkSystem::has_camera(uint8_t camera_id) const
 {
-    uint8_t camera_comp_id = MAV_COMP_ID_CAMERA + (camera_id - 1);
+    uint8_t camera_comp_id = (camera_id == 0) ?
+                             camera_id : (MAV_COMP_ID_CAMERA + (camera_id - 1));
```

Comment:
Does that mean that the first camera has index 1? If yes, that works but should be documented that it is 1 based.

Otherwise, I'd make the default `int camera_id = -1` and check for that, and then use `0` input as the first component.

Reply:
Yes, although it is documented, I think what you're saying is right. Users may feel index `0` as more intuitive.